ui/desktop/electron-app/src/services/rdp-client-manager.js

```diff
@@ -0,0 +1,177 @@
+        if (fs.existsSync('/Applications/Windows App.app')) {
+          return true;
+        }
+
+        // Fallback: Use mdfind for Microsoft Remote Desktop
+        return new Promise((resolve) => {
+          const mdfind = spawn('mdfind', [
+            'kMDItemCFBundleIdentifier == "com.microsoft.rdc.macos"',
+          ]);
+          let output = '';
+
+          mdfind.stdout.on('data', (data) => {
+            output += data.toString();
+          });
+
+          mdfind.on('close', (code) => {
+            const found = code === 0 && output.trim().length > 0;
+            resolve(found);
+          });
+
+          mdfind.on('error', () => resolve(false));
+        });
+      } catch {
+        return false;
+      }
+    },
+  },
+  {
+    value: 'none',
+    isAvailable: () => true,
+  },
+];
+
+class RdpClientManager {
+  /**
+   * Gets all available RDP clients on the current system
+   * @returns {Promise<Array>} Array of available RDP clients with value
+   */
+  async getAvailableRdpClients() {
+    const available = [];
+    for (const client of RDP_CLIENTS) {
+      if (await client.isAvailable()) {
+        available.push({
+          value: client.value,
+        });
+      }
+    }
+    return available;
+  }
+
+  /**
+   * Gets the best default RDP client (first available that's not 'none')
+   * @returns {Promise<string>} Best RDP client value or 'none'
+   */
+  async getBestDefaultRdpClient() {
+    const availableClients = await this.getAvailableRdpClients();
+    const bestClient = availableClients.find(
+      (client) => client.value !== 'none',
+    );
+    return bestClient ? bestClient.value : 'none';
+  }
+
+  /**
+   * Gets the user's preferred RDP client, auto-detecting if not set
+   * @returns {Promise<string>} Preferred RDP client value
+   */
+  async getPreferredRdpClient() {
+    let preferredClient = store.get('preferredRdpClient');
+
+    if (!preferredClient) {
+      // Auto-detect and set the best available client
+      preferredClient = await this.getBestDefaultRdpClient();
+      store.set('preferredRdpClient', preferredClient);
+    }
+    return preferredClient;
+  }
+
+  /**
+   * Sets the user's preferred RDP client
+   * @param {string} preferredClient - The RDP client value to set as preferred
+   */
+  setPreferredRdpClient(preferredClient) {
+    try {
+      if (!preferredClient) {
+        store.set('preferredRdpClient', 'none');
+      } else {
+        store.set('preferredRdpClient', preferredClient);
+      }
+    } catch (error) {
+      log.error('Error setting preferred RDP client', error);
+    }
+  }
+
+  /**
+   * Launches RDP connection with the specified address and port
+   * @param {string} address - Target address
+   * @param {number} port - Target port
+   */
+  async launchRdpConnection(address, port) {
+    if (isWindows()) {
+      // Launch Windows mstsc
+      const mstscArgs = [`/v:${address}:${port}`];
+      const child = spawn('mstsc', mstscArgs, {
```

Comment:
Any reason we don't use one of our own `spawn` helpers?

What's managing this spawned process? As it is, this is a resource leak.